### PR TITLE
DataModel Diff

### PR DESCRIPTION
**Pull request by aly76**
#280

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -426,6 +427,28 @@ EXAMPLE
   $ sfdx sfpowerkit:project:manifest:merge -p project1/path/to/package.xml,project2/path/to/package.xml -d result/package.xml
 ```
 
+### `sfpowerkit:project:datamodel:diff`
+
+Provides an audit history of the metadata change between two commit ID's
```
Comment (azlam-abdulsalam) [inline README.md:432]:
```suggestion
Provides an audit history of the metadata change between two commit ID's
```
Provides an audit history of the metadata change between two commit ID's for data model ( CustomFields, RecordTypes, BusinessProcess)

File: README.md [hunk 2]
```diff
@@ -426,6 +427,28 @@ EXAMPLE
...
+
+```
+USAGE
+  $ sfdx sfpowerkit:project:datamodel:diff -r <string> [-t <string>] [-p <string>] [-d <directory>] [--csv] [--json] [--loglevel
+  trace|debug|info|warn|error|fatal|TRACE|DEBUG|INFO|WARN|ERROR|FATAL]
+
+OPTIONS
+  -d, --outputdir=outputdir                                                         Directory to output the results
+  -p, --packagedirectories=packagedirectories                                       package directories to run diff on
```
Comment (azlam-abdulsalam) [inline README.md:441]:
```suggestion
  -p, --packagedirectories=packagedirectories                                       package directories to run diff on
```
Run diff only for specified package directories

File: package.json [hunk 3]
```diff
@@ -111,6 +116,12 @@
       "sfpowerkit:pool": {
         "description": "Commands to create a pool of scratch orgs"
       },
+      "sfpowerkit:project": {
+        "description": "Commands that operate on the project"
+      },
+      "sfpowerkit:project:datamodel": {
+        "description": "Commands that operate on the data model"
```
Comment (azlam-abdulsalam) [inline package.json:123]:
```suggestion
        "description": "Commands that operate on the data model"
```
progject:datamodeldiff?  any suggestions

File: src/commands/sfpowerkit/project/datamodel/diff.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+      "api_name",
+      "type",
+      "operation",
+      "coordinates",
+      "from",
+      "to"
+    ]);
+    this.ux.log("\n");
+    if (rowsToDisplay.length > 50) {
+      SFPowerkit.log("Displaying output limited to 50 rows", LoggerLevel.WARN);
```
Comment (azlam-abdulsalam) [inline src/commands/sfpowerkit/project/datamodel/diff.ts:189]:
```suggestion
      SFPowerkit.log("Displaying output limited to 50 rows", LoggerLevel.INFO);
```
Info should be enough

File: README.md [hunk 5]
```diff
@@ -51,6 +51,7 @@ $ sfdx plugins:link
     - [`sfpowerkit:project:orgdiff`](#sfpowerkitprojectorgdiff)
     - [`sfpowerkit:project:manifest:diff`](#sfpowerkitprojectmanifestdiff)
     - [`sfpowerkit:project:manifest:merge`](#sfpowerkitprojectmanifestmerge)
+    - [`sfpowerkit:project:datamodel:diff`](#sfpowerkitprojectdatamodeldiff)
```
Comment (azlam-abdulsalam) [inline README.md:54]: Beta?

File: src/commands/sfpowerkit/project/datamodel/diff.ts (hunk 4 above)
Comment (aly76) [reply src/commands/sfpowerkit/project/datamodel/diff.ts:189]: I prefer warning as it shows in a different colour, prompting the user to go check the file for the full results.

Comment (manisfdcsfdx) [reply src/commands/sfpowerkit/project/datamodel/diff.ts:189]: Warn should be fine

File: package.json (hunk 3 above)
Comment (manisfdcsfdx) [reply package.json:123]: progject:datamodel is correct as it is a stream and we will get series of commands under it.

**Pull request merged**